src/lib.rs

```diff
@@ -440,8 +443,7 @@ fn fmt_bar( bytes : &Vec<u64>, max_bytes : u64, width : usize, ascii_flag : bool
         if x > pos {
             total = part;
             part  = bytesi.next().unwrap_or(&0);
-            bars  = ( part * bars ) / total;
-
+            bars = match total { 0 => 0, _ => (part * bars) / total };
```

Comment:
Not super familiar with Rust or the preferred conventions here, so happy to refactor these if there's a more concise way to avoid these issues :slightly_smiling_face: